xarray/core/dataset.py

```diff
@@ -1241,13 +1242,25 @@ def loc(self) -> _LocIndexer:
         """
         return _LocIndexer(self)
 
-    def __getitem__(self, key: Any) -> "Union[DataArray, Dataset]":
+    # FIXME https://github.com/python/mypy/issues/7328
```

Comment:
It looks like this is fixed now and can be removed? Or perhaps we more it below above the third `@overload` and add a comment that `Any` means `list`?

Reply:
I don't think it's fixed. Specifically, mypy can't deal in the signature with an overload of Mapping and Hashable. Curiously however, once you add ``#type: ignore`` to the overloaded signature, the actual type inspection works just fine (see my test script above).